### PR TITLE
Update checks.yml

### PR DESCRIPTION
**Pull request by adam-a-a**
Update CI to check python 3.11 and enable user-mode

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -51,13 +52,13 @@ jobs:
...
         run: |
           pip install -r requirements-dev.txt
-      # - name: Install (noneditable)
-      #   if: matrix.install-mode == 'noneditable'
-      #   run: |
-      #     pip install .
+      - name: Install (noneditable)
+        if: matrix.install-mode == 'noneditable'
+        run: |
+          pip install .
```
Comment (lbianchi-lbl) [inline .github/workflows/checks.yml:58]:
```suggestion
          pip install . && rm -rf *  # delete directory contents after installing to simulate installing from PyPI
```

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -51,13 +52,13 @@ jobs:
...
       - name: Set up IDAES solvers
         run: |
           idaes get-extensions --verbose
       - name: Run pytest
         run: |
-          pytest
\ No newline at end of file
+          pytest
```
Comment (lbianchi-lbl) [inline .github/workflows/checks.yml:64]:
```suggestion
         pip install pytest  # ensure pytest is installed (should do nothing if already present from requirements-dev.txt
          pytest --pyargs watertap_contrib.reflo
```

**Comment by lbianchi-lbl** [comment]
@adam-a-a @kurbansitterley I believe this PR addresses all of the test failures we ran into during the 0.1.0rc0 release process. I've also restored (and improved) the CI checks exercising the non-editable installation so that these types of issues are less likely to slip through next time.

**Comment by adam-a-a** [comment]
I can merge this without reviews, or you two can approve. Any preference?

**Review by lbianchi-lbl — APPROVED** [review]

**Review by kurbansitterley — APPROVED** [review]

**Pull request merged**